source/NVDAHelper.py

```diff
@@ -123,20 +123,31 @@ def nvdaControllerInternal_requestRegistration(uuidString):
 
 
 @WINFUNCTYPE(c_long, c_wchar_p, c_wchar_p)
-def nvdaControllerInternal_reportLiveRegion(text, level):
+def nvdaControllerInternal_reportLiveRegion(text, politeness):

```

Comment:
looking at the ctypes documentation, I can't seem to confirm that the params here `text` and `politeness` won't refer to the original memory. I think they must do, otherwise there would be no way to use them as outparams for instance. 

This will be a problem because they are then used later, after this call completes. Given this is called from `ia2LiveRegions.cpp` and these are temporaries, accessing the memory is an error. 

I'm not 100% sure there isn't some implicit type conversion / adaptation happening in ctypes here. You might consider confirming this by checking their type EG `assert isinstance(text, str)`.

Reply:
Mm, interesting point. I've not seen earlier concerns about this, and given the prototype of the function is added above, I don't think text and politeness won't be anything else than a string.
Also, note that there's also RPC involved in the process. See [this](https://docs.microsoft.com/en-us/windows/win32/rpc/strings) documentation.

Reply:
I think this needs to be resolved still. We don't want ctypes string stand-ins from leaking into the rest of NVDA. Could you tell me the result of adding an assert checking if this is a str. If it is not, it should be converted.

Reply:
I will add assert statements as requested.
However, according to your theory, I assume `nvdaController_cancelSpeech`, `nvdaController_brailleMessage`, `nvdaControllerInternal_logMessage`, `nvdaControllerInternal_inputCompositionUpdate` and `nvdaControllerInternal_inputCandidateListUpdate` will also suffer from this possibly, and I've never seen issues with those.

Reply:
If it is already a python `str` I don't think there is a problem. But if it is a `c_wchar_p` then I'm not so sure, that may be backed by the pointer provided in the call from C++. If we are using that after the function exits, we are moving into the territory of undefined behavior.

So there are a number of 'ifs' involved. I see you added the assert, I assume this means that it is indeed a `str` object, and not just a failing assert?